stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -1089,22 +1089,39 @@ const RealOrComplex = Union{Real,Complex}
     *(A, B::AbstractMatrix, C)
     A * B * C * D
 
-Chained multiplication of 3 or 4 matrices is done in the most efficient order,
-i.e. the order which minimises the total number of scalar multiplications involved,
-based on the sizes of the arrays.
-
-If the last factor is a vector, or the first a transposed vector,
-it is efficient to do these first. In particular `x' * B * y` means `(x' * B) * y`
-for an ordinary colum-major `B`. This is usually faster than `dot(x, B, y)`,
+Chained multiplication of 3 or 4 matrices is done in the most efficient sequence,
+based on the sizes of the arrays. That is, the number of scalar multiplications needed
+for `(A * B) * C` (with 3 dense matrices) is compared to that for `A * (B * C)`
+to choose which of these to execute.
+
+If the last factor is a vector, or the first a transposed vector, then it is efficient
+to deal with these first. In particular `x' * B * y` means `(x' * B) * y`
+for an ordinary colum-major `B::Matrix`. This is often equivalent to `dot(x, B, y)`,
```

Comment:
colum -> column

Reply:
Hm, why not be explicit: "For scalar eltypes, this is equivalent to `dot(x, B, y)`" or something in that direction.

Reply:
Maybe "Unlike dot(), ..."? I don't want to dwell on pinning down exactly what types they agree or disagree on.

Reply:
As an aside, trying to puzzle out what recursive `dot` means in the 3-arg case, is this the intended behaviour? 
```
julia> dot(x,A,y)
7.411848453027886

julia> @tullio _ := x[i][b,a] * A[i,j][b,c] * y[j][c,a]
7.4118484530278845

julia> @tullio _ := tr(x[i]' * A[i,j] * y[j])
7.411848453027887
```
The action on the component matrices looks like a trace of a matrix product.